### PR TITLE
fix multi carrier not displayed

### PR DESCRIPTION
**Pull request by JeromeFarin**
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22583)

### CODE REVIEW

File: classes/checkout/CheckoutDeliveryStep.php [hunk 1]
```diff
@@ -195,7 +195,16 @@ protected function isModuleComplete($requestParams)
     {
         $deliveryOptions = $this->getCheckoutSession()->getDeliveryOptions();
         $currentDeliveryOption = $deliveryOptions[$this->getCheckoutSession()->getSelectedDeliveryOption()];
-        if (!$currentDeliveryOption['is_module']) {
+
+        $isModule = false;
+
+        foreach ($currentDeliveryOption as $option) {
+            if ($option['is_module']) {
+                $isModule = true;
```
Comment (PierreRambaud) [inline classes/checkout/CheckoutDeliveryStep.php:203]:
```suggestion
                $isModule = true;
                break;
```
No need to continue if you found it :)

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Please fill the description and create an issue

**Comment by hibatallahAouadni** [comment]
I think this PR may fix this issue #15369 (as mentioned by @JeromeFarin)

**Comment by PierreRambaud** [comment]
Hi,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

File: classes/Cart.php [hunk 2]
```diff
@@ -2982,11 +3052,55 @@ public function getDeliveryOptionList(Country $default_country = null, $flush =
...
+                            $new_multi_carriers[$key . $id_carrier . ','] = $value;
+                            $new_multi_carriers[$key . $id_carrier . ',']['carrier_list'][$id_carrier] = [
+                                'price_with_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['with_tax'],
+                                'price_without_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['without_tax'],
+                                'instance' => $carriers_instance[$id_carrier],
+                                'package_list' => [$id_package],
+                                'product_list' => $package['product_list'],
+                            ];
+                        }
+                        $multi_carriers = $new_multi_carriers;
```
Comment (MultiplicaliaIT) [inline classes/Cart.php:3088]: I think this code has an error, the line:

$multi_carriers = $new_multi_carriers 

should go after the second foreach is closed and no after the first foreach.

**Comment by Codencode** [comment]
Hi everyone,
why were these changes not taken into consideration?
They fix a problem mentioned in various issues #15369, #15491

**Comment by kpodemski** [comment]
Hi @Codencode 

Yes, it's a complex subject because this PR only touches some parts of the multi-shipping feature. It requires changes in BO, Front office, and emails.

**Pull request closed without merging**